File: fortran_pi_dir/pi.f90
program pi

  implicit none

  double precision  :: step, x, sum, mypi, start, stop
  integer*8         :: num_steps, i
  character(len=32) :: arg

  num_steps = 1000000000

! Get command line args (Fortran 2003 standard)
  if (command_argument_count() > 0) then
     call get_command_argument(1, arg)
     read(arg,*) num_steps
  end if

! Output start message

  write(*,'(A)') "Calculating PI using:"
  write(*,'(A,1I16,A)') "                  ",num_steps, " slices"
  write(*,'(A,1I16,A)') "                  ",1," process"

! Initialise time counter and sum: set step size

  call cpu_time(start)
  sum = 0.
  step = 1.0d0 / num_steps

  do i = 1, num_steps
    x = (i - 0.5d0) * step
    sum = sum + 4.0d0 / (1.0d0 + x*x)
  end do

! Evaluate PI from the final sum value, and stop the clock

  mypi = sum * step
  call cpu_time(stop)

! output value of PI and time taken
! note cpu_time is only specified as being microsecond res

  write(*,'(A,1F12.10,A)') "Obtained value of PI: ", mypi
  write(*,'(A,1F12.5,A)') "Time taken:           ",(stop-start), " seconds"

end program pi

File: python3_pi_dir/pi.py
# Version of the Python Pi example compatible 3 optimised.

# Dr Owain Kenway

import time
import sys


num_steps = 10000000

if len(sys.argv) > 1:
	num_steps = int(sys.argv[1])

print(f"Calculating PI with:\n  {num_steps} slices")

total_sum = 0
step = 1.0 / num_steps

start = time.time()

p = step * sum(4.0/(1.0 + ((i + 0.5) * (i + 0.5) * step * step)) for i in range(num_steps))

stop = time.time()

print(f"Obtained value of PI: {p}")
print(f"Time taken: {stop - start} seconds")
